bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/solcast/handler/SolcastPlaneHandler.java

```diff
@@ -0,0 +1,333 @@
+                            Instant.now().plus(configuration.get().refreshInterval, ChronoUnit.MINUTES),
+                            bridgeHandler.get());
+
+                    // get forecast
+                    Request forecastRequest = httpClient.newRequest(forecastUrl);
+                    forecastRequest.header(HttpHeader.AUTHORIZATION, BEARER + bridgeHandler.get().getApiKey());
+                    ContentResponse crForecast = forecastRequest.send();
+
+                    if (crForecast.getStatus() == 200) {
+                        localForecast.join(crForecast.getContentAsString());
+                        setForecast(localForecast);
+                        updateState(CHANNEL_RAW, StringType.valueOf(forecast.get().getRaw()));
+                        updateStatus(ThingStatus.ONLINE);
+                    } else {
+                        logger.debug("{} Call {} failed {}", thing.getLabel(), forecastUrl, crForecast.getStatus());
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "@text/solarforecast.plane.status.http-status [\"" + crForecast.getStatus() + "\"]");
+                    }
+                } else {
+                    logger.debug("{} Call {} failed {}", thing.getLabel(), currentEstimateUrl, crEstimate.getStatus());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "@text/solarforecast.plane.status.http-status [\"" + crEstimate.getStatus() + "\"]");
+                }
+            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                logger.debug("{} Call {} failed {}", thing.getLabel(), currentEstimateUrl, e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        } // else use available forecast
+        updateChannels(forecast.get());
+        if (Instant.now().isAfter(nextMeasurement.get())) {
+            sendMeasure();
+        }
+        return forecast.get();
+    }
+
+    /**
+     * https://legacy-docs.solcast.com.au/#measurements-rooftop-site
+     */
+    private void sendMeasure() {
+        State updateState = UnDefType.UNDEF;
+        if (persistenceService.isPresent() && powerItem.isPresent()) {
+            ZonedDateTime beginPeriodDT = nextMeasurement.get().atZone(bridgeHandler.get().getTimeZone())
+                    .minusMinutes(MEASURE_INTERVAL_MIN);
+            ZonedDateTime endPeriodDT = nextMeasurement.get().atZone(bridgeHandler.get().getTimeZone());
+            FilterCriteria fc = new FilterCriteria();
+            fc.setBeginDate(beginPeriodDT);
+            fc.setEndDate(endPeriodDT);
+            fc.setItemName(configuration.get().powerItem);
+            Iterable<HistoricItem> historicItems = persistenceService.get().query(fc);
```

Comment:
Not really interaction - just reading. Sadly there's no `averageSince ` for items like provided in rules. That's all I need!

Reply:
See https://github.com/openhab/openhab-addons/pull/13308#discussion_r1432690380

Reply:
Since Solcast removed the tuning feature AFAIK, is there any value in sending actual power values to Solcast? 
If it doesn’t improve the forecast anymore, I don’t see a reason to send that to Solcast, so this code could be removed.

Reply:
Feature removed